src/buffer.c

```diff
@@ -41,6 +41,9 @@ int git_buf_try_grow(
 	if (buf->ptr == git_buf__oom)
 		return -1;
 
+	if (buf->asize == 0 && buf->size != 0)
+		return GIT_EINVALIDSPEC;
```

Comment:
It was indeed about specifications, but we've co-opted it a bit for saying that there was invalid input, which is something I've spoken against in the past.